### PR TITLE
Add sign-in page

### LINKED ISSUE #8: Add authentication
User should be able to create account and log in. It should work with back-ends token based authentication (anttilip/evento-server#9).

### PR DESCRIPTION
**Pull request by anttilip**
Even though back end does not yet use authentication (see anttilip/evento-server#17) I've tested this with a local dev server running and it worked fine. This PR should not be merged until tested with the production server. I also added event cards to My Events page to test.

For cookies I used [js-cookie](https://github.com/js-cookie/js-cookie). Currently cookies store `auth_key` and `user_id`, which are required for most API requests.

Closes #8

### CODE REVIEW

File: evento/src/views/SignInPage/index.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		}).then(response => {
+			if (response.status === 200) {
+				return response.json();
+			}
+			this.setState({ errorMessage: 'Invalid credentials' });
+		}).then(authKey => {
+			Cookie.set('auth_token', authKey.auth_token);
+
+			// User id needs to be fetched and saved to cookie
+			fetch('/users')
```
Comment (JaakkoLipsanen) [inline evento/src/views/SignInPage/index.js]: We shouldn't do this. If the '/authentication' request succeeds, it should return the user in the response. Makes much more sense than this

File: evento/src/views/SignInPage/index.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+		return (
+			<form className='SignInPage' onSubmit={(e) => this.handleSubmit(e)}>
+				<p>{this.state.errorMessage}</p>
+				<label>
+					Email:<br/>
+					<input type="text" value={this.state.email} onChange={(evt) => this.setState({email: evt.target.value})} />
+					<br/>Password:<br/>
+					<input type="password" value={this.state.password} onChange={(evt) => this.setState({password: evt.target.value})} />
+				</label><br/>
+				<input type="submit" value="Submit" />
```
Comment (JaakkoLipsanen) [inline evento/src/views/SignInPage/index.js]: Why not just button-tag since we are not using the auto-POST? Much clearer

File: evento/src/views/MyEvents/index.js [hunk 3]
```diff
@@ -1,14 +1,38 @@
...
+	constructor() {
+		super();
+		this.state = { events: [] };
+	}
+
+	componentDidMount() {
+		const userId = Cookie.get('userId');
+		const authToken = Cookie.get('auth_token');
+
+		fetch(`/users/${userId}/events`, {
```
Comment (JaakkoLipsanen) [inline evento/src/views/MyEvents/index.js]: What happens if there is no userId or auth_token cookies?

File: evento/src/views/SignInPage/index.js (hunk 1 above)
Comment (anttilip) [inline evento/src/views/SignInPage/index.js]: Yeah but currently API does not do that yet so that is still needed.

Comment (JaakkoLipsanen) [inline evento/src/views/SignInPage/index.js]: Sure, but we should implement it in the backend now instead of pushing hacks like that to production

Comment (anttilip) [inline evento/src/views/SignInPage/index.js]: Of course. This PR is not intended to be merged to master yet. Even authentication is not yet implemented in backend so none of this would work.

File: evento/src/views/MyEvents/index.js (hunk 3 above)
Comment (anttilip) [inline evento/src/views/MyEvents/index.js]: fetch is directed to `/users/undefined/events` and Authorization header undefined. So in essence the fetch fails and state.events will stay empty.

Comment (JaakkoLipsanen) [inline evento/src/views/MyEvents/index.js]: Meh. I'd say that move all of the code inside componentDidMount to "fetchEvents" or something and then do an early exit if the cookies are undefined.

File: evento/package.json [hunk 4]
```diff
@@ -19,14 +20,11 @@
     "build-css": "node-sass src/ -o src/",
     "watch-css": "npm run build-css -- -q && node-sass src/ -o src/ --watch --recursive",
     "start-js": "react-scripts start",
-
```
Comment (JaakkoLipsanen) [inline evento/package.json]: :(

File: evento/src/views/App/index.js [hunk 5]
```diff
@@ -13,8 +14,9 @@ class App extends Component {
 			<div className="App">
 				<Router>
 					<Switch>
-						<Route exact path='/' component={MainPage} />
+						<Route exact path='/signin' component={SignInPage} />
 						<Route exact path='/event/:eventId' component={EventPage} />
+						<Route path='/' component={MainPage} />
```
Comment (JaakkoLipsanen) [inline evento/src/views/App/index.js]: Why did you move this here? And remove exact?

**Comment by JaakkoLipsanen** [comment]
Looks ok

File: evento/src/views/App/index.js (hunk 5 above)
Comment (anttilip) [inline evento/src/views/App/index.js]: `<Route path='/' component={MainPage} />` does not match `/events` , which is the path for MyEvents. Maybe better way would be something like `path='/' or '/events'`.

Comment (JaakkoLipsanen) [inline evento/src/views/App/index.js]: Oh right :/ Maybe we should do two routes, one for '/' and one for '/events' with both having MainPage as the component. It would be a bit clearer since A: invalid routes would not be directed there (in your code 404 page is never fired?) and B: all routes would be explicitly defined in here so it would be easier to check all the valid routes.

**Comment by JaakkoLipsanen** [comment]
Closes #34
